patches/minecraft/net/minecraft/entity/Entity.java.patch

```diff
@@ -31,18 +31,20 @@
        this.field_200606_g = p_i48580_1_;
        this.field_70170_p = p_i48580_2_;
        this.field_213325_aI = p_i48580_1_.func_220334_j();
-@@ -213,7 +216,9 @@
+@@ -213,7 +216,11 @@
        this.field_70180_af.func_187214_a(field_189655_aD, false);
        this.field_70180_af.func_187214_a(field_213330_X, Pose.STANDING);
        this.func_70088_a();
 -      this.field_213326_aJ = this.func_213316_a(Pose.STANDING, this.field_213325_aI);
-+      this.field_213326_aJ = getEyeHeightForge(Pose.STANDING, this.field_213325_aI);
++      net.minecraftforge.event.entity.EntityEvent.Size sizeEvent= getSizeForge(Pose.STANDING, this.field_213325_aI);
```

Comment:
I am not sure about this part.
The old eyeHeight method was placed in the constructor as well, so I kept it here.
However, I do not think this is ideal. If mods decide to change an entity's size/eyeHeight, they need something to base their decision upon. But at this point during entity construction neither capabilities nor held items nor anything else is available.
This means a) the event handler has to check if the entity is properly constructed and b) the mod has to trigger a size update anyway.
Only scenario this might be useful might be a mod which e.g. simply scales all cows to 200%. But that could also be done without the constructor patch, by just calling recalculateSize in EntityJoinWorld.

Therefore, I would suggest removing this ugly patch, making it much easier to use.

Reply:
The issue is that we have to have SOME default value in the constructor or somewhere that is always called before any other methods in the entity is called because eyeHeight needs to be a sane value until the next time resize is called.

Reply:
I would simply use the vanilla methods for size and eyeHeight. Thereby they are at sane values, and if desired a mod can still force a size recalculation once the entity is fully constructed.